tests/cpp/target_test.cc

```diff
@@ -448,6 +449,21 @@ TEST(TargetCreation, LLVMCommandLineSaveRestore) {
   ASSERT_FALSE(info.MatchesGlobalState());
 }
 
+TEST(TargetCreation, DetectSystemTriple) {
```

Comment:
I think we should keep this test outside the `#ifdef TVM_INFO_USE_LLVM` scope so that we get coverage when LLVM is not available. We can use the skip functionality provided by gtest before making a call to LLVM dependent functions (see comment below). That way we're _somewhat_ able to test the changes made in `cpu.cc` this PR makes.

... I say somewhat because the "gpu other" build (the only build in CI that uses USE_LLVM=OFF) doesn't run the cpp test suite. Building and running the tests have been commented out: https://github.com/apache/tvm/blob/a15fd1d6a21e9bedb622ffb9fd025c50624d734d/ci/jenkins/templates/gpu_jenkinsfile.groovy.j2#L57. There is a hint here: https://github.com/apache/tvm/pull/13335/files#r1023416998. But we should probably investigate this further in a separate PR, so we don't block this fix.